compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt

```diff
@@ -158,11 +158,49 @@ internal abstract class ComposeBridge(
         }
     }
 
-    protected val sceneDimension: Dimension
-        get() = Dimension(
-            (scene.contentSize.width / component.density.density).toInt(),
-            (scene.contentSize.height / component.density.density).toInt()
+    /**
+     * Provides the maximum constraints where ComposeScene can be rendered.
+     * Dimensions' sum of all displays is used for that.
+     * Since we cannot show more than on all the screens.
+     *
+     * See [scenePreferredSize]
+     */
+    private fun getMaxConstraints(): Constraints {
+        val ge = GraphicsEnvironment.getLocalGraphicsEnvironment()
+        val deviceBounds = ge.screenDevices.map { it.defaultConfiguration.bounds }
+
+        val totalWidth = deviceBounds.sumOf { it.width }
+        val totalHeight = deviceBounds.sumOf { it.height }
+
+        val density = component.density.density
+        return Constraints(
+            maxWidth = (totalWidth * density).toInt(),
+            maxHeight = (totalHeight * density).toInt()
         )
```

Comment:
The only reason is the ability to use Lazy* or not. Since when you try to measure e.g. LazyColumn in Infinity height it will throw an exception.
And it won't be so easy to understand what's the reason of the error. 
Also, it can break current applications

Reply:
But I am for Inifinity + handle exception, check exception text and provide additional information, that preferred size is calculated -> set preferred size/add specific size for Lazy*

Reply:
The only thing that I worry about is breaking existing applications. For sure, only not properly written applications will be broken

Reply:
We could measure like before in the case of a fully Compose app (WindowComposeBridge). Or decide it's not a valid case to have a window size of `Dp.Unspecified` and a content with no intrinsic size. @igordmn what do you think?

Reply:
> We could measure like before in the case of a fully Compose app

We actually measured the same way as in this PR (see `Window.setSizeImpl(size: DpSize)`) for pure Compose applications. Now we use this logic for ComposeWindow/ComposePanel as well.

>  it's not a valid case to have a window size of Dp.Unspecified and a content with no intrinsic size

It is not a valid case, we can't calculate prefered size of LazyList in any meaningful way. 

Actually, I have a question, is there a case in Swing for that? If we don't call `pack()` will `getPreferredSize()` be called? If not, I would just pass Infinity and throw an exception (without handling it).

Reply:
`If we don't call pack() will getPreferredSize()` preferredSize can be called anytime see the related issue:  https://github.com/JetBrains/compose-multiplatform/issues/3913

Reply:
This case is valid - we use preferred size to determine the size of popup.

But is there any functionality that calls `getPreferredSize` and ignores its result? If there is no any, we can safely just pass Infinity as constraints, and throw an exception for LazyList

Reply:
I'd provide one more usecase then: completion popup with just a few items. It will always be a LazyColumn, but it hcan have thousands or just a few items and when there are only a few of themm then preferredSize should return the exact size, not throwing an exception I guess.
But may be it can be properly handled from the user-side?

Reply:
> But may be it can be properly handled from the user-side?

In this case, the user should specify the max size of the lazy list. Ther won't be an exception then.

Reply:
Good point, agree with Infinite constraints then